prow/plugins/approve/approvers2/owners.go

```diff
@@ -725,3 +879,144 @@ func getGubernatorMetadata(toBeAssigned []string) string {
 	}
 	return ""
 }
+
+// wildcardPathMatch return true if the targetPath matches the given pattern
+// A '*' as a directory will recursively match all sub directories
+//    Example: file a/b1/b2/c.go matches the pattern a/*/c.go
```

Comment:
This is surprising to me, not sure I like the syntax. It doesn't match https://pkg.go.dev/path/filepath#Match behavior.  I'm used to `**` signifying recursive matching (though go doesn't natively support this)

Reply:
The initial implementation did the `**` approach. Some folks suggested that using `*` might be easier instead of having to switch between `*` for any match and `**` for recursive match. 
cc @nikhita 

I am +1 to switching back if that is the majority's preference.

Reply:
> The initial implementation did the ** approach. Some folks suggested that using * might be easier instead of having to switch between * for any match and ** for recursive match.

Honestly, I can't recall why we had decided on the `*` approach. If `**` works, let's just go with it. Would love to merge and iterate.

Reply:
I would prefer `**`

In terms of merge and iterate, I would prefer merging a PR that _just_ does per-file checking, and then we can iterate with a later PR that adds `/approve files foo` support.  Otherwise, I would rather see the `**` syntax fixed first.